lib/engine/part/path.rb

```diff
@@ -65,13 +90,24 @@ def select(paths)
         on.keys.select { |p| on[p] == 1 }
       end
 
-      def walk(skip: nil, visited: nil, on: nil)
+      def walk(skip: nil, jskip: nil, visited: nil, on: nil)
         return if visited&.[](self)
 
         visited = visited&.dup || {}
         visited[self] = true
         yield self, visited
 
+        [@a, @b].each do |j|
```

Comment:
i thin you could replace this with just grabbing @junction because it is an instance variable of path

Reply:
Yep